lib/sinatra/base.rb

```diff
@@ -1005,7 +1005,8 @@ def process_route(pattern, keys, conditions, block = nil, values = [])
       values += match.captures.map! { |v| force_encoding URI_INSTANCE.unescape(v) if v }
 
       if values.any?
-        original, @params = params, params.merge('splat' => [], 'captures' => values)
+        original = @params
+        @params = @params.merge('splat' => [], 'captures' => values)  # makes a copy of @params
```

Comment:
Ah, it didn't occur to me that you were commenting the underlying Ruby rather than your change. The difference between `merge` and `merge!` is well-known, and the comment is confusing in this context. I think this would be clearer if you removed the comment.